### PR TITLE
Add physical Coordinate support to SpotFindingResults 

### PR DESCRIPTION
**Pull request by shanaxel42**
This PR adds physical coordinate information to the SpotResultsClass. This is needed for when the coordinates get transferred during decoding.

Depends on #1489 #1515 

See it in action in #1518

### CODE REVIEW

File: starfish/core/intensity_table/intensity_table_coordinates.py [hunk 1]
```diff
@@ -1,46 +1,63 @@
...
-def transfer_physical_coords_from_imagestack_to_intensity_table(
-        image_stack: ImageStack, intensity_table: IntensityTable
-) -> IntensityTable:
+def transfer_physical_coords_to_intensity_table(intensity_table: IntensityTable,
+                                                image_stack: Optional[ImageStack] = None,
+                                                spots: Optional[SpotFindingResults] = None,
+                                                ) -> IntensityTable:
     """
-    Transfers physical coordinates from an Imagestack's coordinates xarray to an intensity table
+    Transfers physical coordinates from either an Imagestack or SpotFindingResults to an
```
Comment (ttung) [inline starfish/core/intensity_table/intensity_table_coordinates.py]: Long term, will there be any workflows that extract physical coordinates from an ImageStack?  Or is that a stopgap measure until SpotFindingResults is produced everywhere?

File: starfish/core/types/_spot_finding_resutls.py [hunk 2]
```diff
@@ -12,7 +15,7 @@ class SpotFindingResults:
     SpotAttributes.
     """
 
-    def __init__(self, spot_attributes_list: Optional[List[Tuple]] = None):
+    def __init__(self, imagestack, spot_attributes_list: Optional[List[Tuple]] = None):
```
Comment (ttung) [inline starfish/core/types/_spot_finding_resutls.py]: is it plausible that datasets get so big that we need to spread spot finding across multiple machines?  If so, having the imagestack as part of spotfinding's results may make it challenging to persist.

It _might_ make sense to store the coordinates from the imagestack, but not necessarily the actual imagestack (as that's really what you need anyway).

@ambrosejcarr, @dganguli

File: starfish/core/intensity_table/intensity_table_coordinates.py (hunk 1 above)
Comment (shanaxel42) [reply starfish/core/intensity_table/intensity_table_coordinates.py]: Yea I think we'll be able to remove the imagestack option once the full refactors done

File: starfish/core/intensity_table/intensity_table_coordinates.py [hunk 3]
```diff
@@ -1,46 +1,63 @@
...
 from starfish.core.imagestack.imagestack import ImageStack
 from starfish.core.intensity_table.intensity_table import IntensityTable
-from starfish.core.types import Axes, Coordinates, Features
+from starfish.core.types import Axes, Coordinates, Features, SpotFindingResults
 
 
-def transfer_physical_coords_from_imagestack_to_intensity_table(
-        image_stack: ImageStack, intensity_table: IntensityTable
-) -> IntensityTable:
+def transfer_physical_coords_to_intensity_table(*, intensity_table: IntensityTable,
```
Comment (ttung) [inline starfish/core/intensity_table/intensity_table_coordinates.py]: may be cleaner as:

```
def transfer_physical_coords_to_intensity_table(
        *,
        intensity_table: IntensityTable,
        image_stack: Optional[ImageStack] = None,
        spots: Optional[SpotFindingResults] = None,
) -> IntensityTable:
    """
```

File: starfish/core/intensity_table/intensity_table_coordinates.py [hunk 4]
```diff
@@ -1,46 +1,63 @@
...
-    # TODO shanaxel42 consider refactoring pixel case where were can just reshape from Imagestack
 
     pairs = (
         (Axes.X.value, Coordinates.X.value),
         (Axes.Y.value, Coordinates.Y.value),
         (Axes.ZPLANE.value, Coordinates.Z.value)
     )
 
+    if image_stack is None and spots is None:
+        raise ValueError("Must provide either SpotFindingResults or ImageStack to "
```
Comment (ttung) [inline starfish/core/intensity_table/intensity_table_coordinates.py]: this is plausibly more wrapped than necessary.

File: starfish/core/intensity_table/intensity_table_coordinates.py [hunk 5]
```diff
@@ -1,46 +1,63 @@
...
+                         "calculate coordinates from")
+
+    coord_ranges: Mapping[str, xr.DataArray]
```
Comment (ttung) [inline starfish/core/intensity_table/intensity_table_coordinates.py]: As I discovered last night, the newest version of xarray will probably want this as:

```suggestion
    coord_ranges: Mapping[Hashable, xr.DataArray]
```

File: starfish/core/intensity_table/intensity_table_coordinates.py [hunk 6]
```diff
@@ -1,46 +1,63 @@
...
     """
-    # TODO shanaxel42 consider refactoring pixel case where were can just reshape from Imagestack
 
     pairs = (
         (Axes.X.value, Coordinates.X.value),
         (Axes.Y.value, Coordinates.Y.value),
         (Axes.ZPLANE.value, Coordinates.Z.value)
     )
 
+    if image_stack is None and spots is None:
```
Comment (ttung) [inline starfish/core/intensity_table/intensity_table_coordinates.py]: consider doing the xor.  you really just want one specified.  both is also bad.

**Pull request merged**